tests/v1/sample/test_rejection_sampler.py

```diff
@@ -198,6 +198,33 @@ def test_empty_sequence(rejection_sampler):
     assert torch.equal(output, expected)
 
 
+def test_nan_logits(rejection_sampler):
+    """Test handling nan values in the target logits"""
+    spec_tokens: list[list[int]] = [[0]]
+    output_tokens = [[0, 0]]  # Just the bonus token
```

Comment:
right now, all the tokens are 0 which gives the impression that they are related in different places. However, they are not. Can we instead change it to
```
spec_tokens: list[list[int]] = [[1]]
output_tokens = [[2, 3]]  # Just the bonus token
```

The bonus token defined later doesn't matter since the draft token will be rejected and be sampled as 0 so the above non-0 token ids will make this apparent. 

IIUC, no matter what the spec_tokens and output_tokens and bonus tokens are, if the logits are nan then it will always return `[0, PLACEHOLDER_TOKEN_ID]`